### PR TITLE
WIP: 🐛 Clean removal of output files

### PR DESCRIPTION
**Pull request by giancarloromeo**
## What do these changes do?
**TL;DR**
Removing (output) files referenced by Project Nodes, leaves the system in an inconsistent state.

Currently, the WebServer is the 1st responder and forwards the deletion requests to the Storage service, which deletes physically the file(s).
The Projects Nodes are not updated accordingly and Director v2 is not informed about the changes.

---
### Proposal
To maintain decoupling while ensuring consistency between the storage and project services, I recommend using an Event-Driven Architecture:

- Event Broker (e.g., RabbitMQ):
  - Introduce an event broker as a central communication channel.
  - When a file is deleted from the storage service, it publishes a `FileDeleted` event with relevant metadata.
- Storage Service:
  - Handles the physical deletion of files in S3.
  - Keeps a local counter for each file for performance and basic deletion decisions.
  - Publishes the `FileDeleted` event once the deletion is successful.
  - Subscribes to `FileReferenced` / `FileDereferenced` events from the event broker.
  - When a `FileReferenced` / `FileDereferenced` event is received, it updates the **soft reference count** associated with that file.
- Project Service:
  - Publishes `FileReferenced` / `FileDereferenced` events once a file is referenced/dereferenced.
  - Subscribes to `FileDeleted` events from the event broker.
  - When a `FileDeleted` event is received, it verifies if the file is referenced in any project and updates its data (e.g., mark the file as deleted, remove reference from Nodes).

### Event Types Overview

| Event Types   | Producer(s)     | Consumer(s)           |
|------------------|-----------------|-----------------------|
| `FileDeleted`    | Storage service | Projects service, ... |
| `FileReferenced` | Projects service, ... | Storage service |
| `FileDereferenced` | Projects service, ... | Storage service |

It would be nice to have the PR #7010 in as well, since it cleans the Project Nodes persistence in the DB.

## Related issue/s

- Resolves: #6963 
- Relates to: #5743
- Relates to: #7010

## Dev-ops checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

**Comment by giancarloromeo** [comment]
The issue is that we have 2 separate services (`storage` and `webserver/projects`) that are not consistent. Whenever a request of file deletion is received from the UI, the file is deleted from the `storage` but the project/s referencing it is/are not updated.

File: services/web/server/src/simcore_service_webserver/projects/_projects_db.py [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 # NOTE: MD: I intentionally didn't include the workbench. There is a special interface
 # for the workbench, and at some point, this column should be removed from the table.
-# The same holds true for access_rights/ui/classifiers/quality, but we have decided to proceed step by step.
+# The same holds true for ui/classifiers/quality, but we have decided to proceed step by step.
```
Comment (giancarloromeo) [inline services/web/server/src/simcore_service_webserver/projects/_projects_db.py:19]: `access_rights` migrated already
